dhis-2/dhis-services/dhis-service-analytics/src/main/java/org/hisp/dhis/analytics/event/data/programindicator/DefaultProgramIndicatorSubqueryBuilder.java

```diff
@@ -171,7 +172,7 @@ private String getWhere( AnalyticsType outerSqlEntity, ProgramIndicator pi, Rela
             }
         }
 
-        return " WHERE " + condition;
+        return isNotBlank( condition ) ? " WHERE " + condition : condition;
```

Comment:
This was a small bug I found during my tests, where using my local data, the `condition` was empty. In such a situation, the return would be a `WHERE` without `condition`, and this was causing an invalid SQL statement exception.